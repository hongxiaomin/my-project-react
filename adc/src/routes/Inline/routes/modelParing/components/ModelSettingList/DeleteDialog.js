import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import { deleteModel } from '../../modules/ModelParing'
import classes from './Dialog.scss'

type Props = {
  dataDetail: Number,
  delOpen: Boolean,
  toggleDel: Function,
  dispatch: Function
};

export class DeleteDialog extends React.Component {


  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    this.props.dispatch(deleteModel(this.props.dataDetail.original.configId))
    this.props.toggleDel()
  }
  props: Props;
  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.delOpen}
            onClose={this.props.toggleDel}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}>Confirmation</h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              <p className={classes.cn}>{'Are you certain that you want to delete this model pairing?'}</p>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleDel}>
                  Cancel
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  Save
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

export default connect()(DeleteDialog)
