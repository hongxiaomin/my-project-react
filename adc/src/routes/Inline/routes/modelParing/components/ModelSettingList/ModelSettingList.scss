.top-left{
  display: flex;
  justify-content: space-between;
  height: 50px;
  width: 100%;
  line-height: 50px;
  font-size: 16px;
  padding: 0px 30px;
  background-color: #f7f7f7;
  // border: 1px solid #d7d7d7;
  border-radius: 4px;
  margin-top: 20px;
}
.button {
  height: 30px;
  margin-top: 10px;
}
.cn{
    text-align: center;
}
.dialog {
  width:400px;
  padding:2px;
  background: #F7F7F7;
}
.headerTitle {
  margin: 10px 0px 10px 15px;
  font-size: 16px;
}

.headerID {
  margin-left: 15px;
  font-size: 18px;
  padding-bottom: 16px;
}
.dialogBody {
  background: #fff;
  color: #000;
}
.filter {
  border-bottom: solid 1px #C9C8C9;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height:65px;
  line-height:60px;
  margin:0px 10px;
  font-size: 14px;
}
.select {
  cursor: pointer;
  outline: 0;
  border: 0;
  background: inherit;
  box-shadow: none;
  margin: 0;
  border: 1px solid #D9D9D9;
  height: 30px;
  width: 160px;
  border-radius: 3px;
}
.btnGroup {
  float: right;
  margin:17px 0px;
}
.btnEdit {
  background: #0087DC;
  color: #fff;
  margin: 0px 10px 0px 15px;
  width: 80px;
  height: 35px;
  font-size: 14px;
}
.btnCancel {
  background: #fff;
  border: 1px solid #0087DC !important;
  color: #0087DC !important;
  width: 80px;
  height: 35px;
  font-size: 14px;
}
.hardCodeText {
  text-align: center;
}
.isContinuous{
  position: absolute;
  bottom: 13px;
  left: 20px;
}
.filter {
  color: #5f5f5f;
  border-bottom: solid 1px #d7d7d7;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.select {
  cursor: pointer;
  outline: 0;
  border: 0;
  background: inherit;
  box-shadow: none;
  width: auto;
  margin: 0;
  direction: rtl;
}
 .rt-tr.-odd {
  display: flex !important;
  align-items: center !important;
}
