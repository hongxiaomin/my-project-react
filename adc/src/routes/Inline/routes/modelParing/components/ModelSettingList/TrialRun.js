import React from 'react'
import PropTypes from 'prop-types'
import { isEmpty, isNumber } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import { updateProcessProduct, getModelSettingList } from '../../modules/ModelParing'
import classes from './Dialog.scss'

export class TrialRun extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit() {
    let tmp
    if (!isNumber(this.props.dataDetail.original.productId)) {
      tmp = {
        modelId: this.props.dataDetail.original.modelId.toString(),
        processId: this.props.dataDetail.original.processId.toString(),
        configId: this.props.dataDetail.original.configId.toString(),
        isTrialRun: !this.props.dataDetail.original.isTrialRun,
      }
    } else {
      tmp = {
        modelId: this.props.dataDetail.original.modelId.toString(),
        processId: this.props.dataDetail.original.processId.toString(),
        configId: this.props.dataDetail.original.configId.toString(),
        productId: this.props.dataDetail.original.productId.toString(),
        isTrialRun: !this.props.dataDetail.original.isTrialRun,
      }
    }

    this.props.dispatch(updateProcessProduct(tmp, () => {
      this.props.dispatch(getModelSettingList({ queryType: 'list', ...this.props.searchParam }))
    }))
    this.props.trialRun()
  }

  render() {
    return (
      <div className={classes['Dialog-container']}>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.trialRunOpen}
            onClose={this.props.trialRun}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={classes.headerTitle}>Confirmation</h6>
            </div>
            <div className={classes.dialogBody}>
              <br />
              {
                !this.props.dataDetail.original.isTrialRun ?
                  <p className={classes.cn}>{'Do you set the model as trial run?'}</p> :
                  <p className={classes.cn}>{'Do you cancel set the model as trial run?'}</p>
              }

              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.trialRun}>
                  Cancel
                </Button>
                <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                  Save
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

TrialRun.propTypes = {
  dataDetail: PropTypes.object.isRequired,
  searchParam: PropTypes.object.isRequired,
  trialRunOpen: PropTypes.bool.isRequired,
  trialRun: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
}

export default connect()(TrialRun)
