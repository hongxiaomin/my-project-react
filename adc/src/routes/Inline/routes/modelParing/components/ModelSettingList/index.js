import ModelSettingList from './ModelSettingList'

export default ModelSettingList
