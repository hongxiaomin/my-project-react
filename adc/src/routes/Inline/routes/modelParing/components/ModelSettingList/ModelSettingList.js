import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
// import { assign } from 'lodash'
// Import React Table
import { connect } from 'react-redux'
import { Icon } from 'antd'
import ReactTable from 'components/ADCTable'
import EditDialog from '../EditDialog'
import Delete from './DeleteDialog'
import TrialRun from './TrialRun'
import classes from './ModelSettingList.scss'
import { getModelSettingList, CHANGE_STATUS } from '../../modules/ModelParing'

export class ModelSettingList extends React.Component {
  constructor() {
    super()
    this.state = {
      highlightedIndex: -1,
      totalPages: null,
      loading: true,
      editOpen: false,
      delOpen: false,
      trialRunOpen: false,
      dataDetail: {},
    }
    this.trialRun = this.trialRun.bind(this)
    this.fileUpload = this.fileUpload.bind(this)
    this.del = this.del.bind(this)
    this.toggleEdit = this.toggleEdit.bind(this)
  }

  componentDidMount() {
    const tmp = {
      queryType: 'list',
    }
    this.props.dispatch(getModelSettingList(tmp))
  }

  toggleEdit(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        editOpen: !this.state.editOpen,
      })
      this.props.dispatch({
        type: CHANGE_STATUS,
        payload: false,
      })
    }
  }
  trialRun(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        trialRunOpen: !this.state.trialRunOpen,
      })
    }
  }
  del(data) {
    return () => {
      this.setState({
        ...this.state,
        dataDetail: data,
        delOpen: !this.state.delOpen,
      })
    }
  }
// 导入功能实现
  fileUpload = () => this.file.click()

  props: Props

  render() {
    const { intl } = this.props

    return (
      <div style={{ margin: '30px 40px' }}>
        <div className={classes['top-left']}>
          <span className={classes.productGlass}>
            <WrapperFormattedMessage id='modelPairing.modelParing.list' />
          </span>
        </div>
        <ReactTable
          data={this.props.modelSettingList}
          defaultPageSize={10}
          className="-striped -highlight"
          columns={[
            {
              Header: intl.formatMessage({ id: 'processId' }),
              accessor: 'processName',
              width: 200,
              style: {
                textAlign: 'center',
                // verticalAlign: 'center',
              },
            },
            {
              Header: intl.formatMessage({ id: 'productId' }),
              accessor: 'productName',
              width: 200,
              style: {
                textAlign: 'center',
              },
            },
            {
              Header: intl.formatMessage({ id: 'serviceManagement.dialogFilter.model' }),
              accessor: 'modelName',
              width: 200,
              style: {
                textAlign: 'center',
              },
            },
            {
              Header: intl.formatMessage({ id: 'labelingTool.eda.defectCode' }),
              id: 'defectCode',
              accessor: d => d.defectType.join('/ '),
              style: {
                textAlign: 'center',
                whiteSpace: 'normal',
              },
            },
            {
              Header: intl.formatMessage({ id: 'serviceManagement.queueList.action' }),
              width: 200,
              accessor: '',
              style: {
                textAlign: 'center',
                padding: '5px',
              },
              Cell: data => (
                <div>
                  {
                    data.original.isTrialRun ?
                      <div style={{ display: 'flex', justifyContent: 'space-around' }}>
                        <button
                          type='button'
                          onClick={this.toggleEdit(data)}
                          className={'pt-button pt-minimal pt-icon-edit'}
                          title={intl.formatMessage({ id: 'serviceManagement.serviceCard.edit' })}
                        />
                        <button
                          type='button'
                          onClick={this.trialRun(data)}
                          className={'pt-button pt-minimal'}
                          title={intl.formatMessage({ id: 'modelPairing.modelParing.cancelTrialRun' })}
                        >
                          <Icon type="star" style={{ color: 'orange' }} />
                        </button>
                        <button
                          type='button'
                          onClick={this.del(data)}
                          className={'pt-button pt-minimal pt-icon-delete'}
                          title={intl.formatMessage({ id: 'delete' })}
                        />
                      </div> :
                      <div style={{ display: 'flex', justifyContent: 'space-around' }}>
                        <button
                          type='button'
                          onClick={this.toggleEdit(data)}
                          className={'pt-button pt-minimal pt-icon-edit'}
                          title={intl.formatMessage({ id: 'edit' })}
                        />
                        <button
                          type='button'
                          onClick={this.trialRun(data)}
                          className={'pt-button pt-minimal'}
                          title={intl.formatMessage({ id: 'modelPairing.modelParing.setAsTrialRun' })}
                        >
                          <Icon type="star" />
                        </button>
                        <button
                          type='button'
                          onClick={this.del(data)}
                          className={'pt-button pt-minimal pt-icon-delete'}
                          title={intl.formatMessage({ id: 'delete' })}
                        />
                      </div>
                   }
                </div>
              ),
            },
          ]}
          getTrProps={() => ({
            style: {
              height: '35px',
              lineHeight: '20px',
            },
          })}
        />
        {
          this.state.editOpen &&
          <EditDialog
            toggleEdit={this.toggleEdit()}
            modelList={this.props.modelList}
            dataDetail={this.state.dataDetail}
            editOpen={this.state.editOpen}
          />
        }
        <TrialRun
          trialRun={this.trialRun()}
          dataDetail={this.state.dataDetail}
          trialRunOpen={this.state.trialRunOpen}
          searchParam={this.props.searchParam}
        />
        <Delete
          toggleDel={this.del()}
          dataDetail={this.state.dataDetail}
          delOpen={this.state.delOpen}
        />
      </div>
    )
  }
}

ModelSettingList.propTypes = {
  dispatch: PropTypes.func.isRequired,
  modelSettingList: PropTypes.array.isRequired,
  modelList: PropTypes.array.isRequired,
  searchParam: PropTypes.object.isRequired,
}

export default connect(
  ({ auth, modelParing, intl: { locale } }) => ({
    locale,
    token: auth.token,
    modelSettingList: modelParing.modelSettingList,
  }),
)(injectIntl(ModelSettingList))
