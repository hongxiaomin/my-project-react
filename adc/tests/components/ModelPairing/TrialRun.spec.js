import React from 'react'
import { TrialRun } from 'routes/Inline/routes/ModelParing/components/ModelSettingList/TrialRun'

describe('(Component) TrialRun', () => {
  const trialRunOpen = false
  const dispatch = jest.fn()
  const trialRun = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <TrialRun
        dataDetail={{ original: { isTrialRun: false } }}
        trialRunOpen={trialRunOpen}
        trialRun={trialRun}
        dispatch={dispatch}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render isTrialRun=true', () => {
    const wrapper = shallow(
      <TrialRun
        dataDetail={{ original: { isTrialRun: true } }}
        trialRunOpen={trialRunOpen}
        trialRun={trialRun}
        dispatch={dispatch}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render productid=null', () => {
    const wrapper = shallow(
      <TrialRun
        dataDetail={{ original: 
          { isTrialRun: true, productId: '', 
            modelId: { toString: jest.fn()},
            processId: { toString: jest.fn()},
            configId: { toString: jest.fn()},
            productId: { toString: jest.fn()}
          } }}
        trialRunOpen={trialRunOpen}
        trialRun={trialRun}
        dispatch={dispatch}
      />,
    )
    wrapper.find('Button').at(1).simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
  it('render productid !== null', () => {
    const wrapper = shallow(
      <TrialRun
      dataDetail={{ original: 
        { isTrialRun: true, productId: 1, 
          modelId: { toString: jest.fn()},
          processId: { toString: jest.fn()},
          configId: { toString: jest.fn()},
        } }}
        trialRunOpen={trialRunOpen}
        trialRun={trialRun}
        dispatch={dispatch}
      />,
    )
    wrapper.find('Button').at(1).simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
})
