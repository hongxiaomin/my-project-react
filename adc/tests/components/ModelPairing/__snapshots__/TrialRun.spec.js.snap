// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TrialRun render isTrialRun=true 1`] = `
<div
  className="Dialog-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        Confirmation
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        Do you cancel set the model as trial run?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;

exports[`(Component) TrialRun render normally 1`] = `
<div
  className="Dialog-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        Confirmation
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        Do you set the model as trial run?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;

exports[`(Component) TrialRun render productid !== null 1`] = `
<div
  className="Dialog-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        Confirmation
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        Do you cancel set the model as trial run?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;

exports[`(Component) TrialRun render productid=null 1`] = `
<div
  className="Dialog-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        Confirmation
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        Do you cancel set the model as trial run?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;
